From c8baa4db8e3913b917f652d345570ddf19a16d13 Mon Sep 17 00:00:00 2001
From: Ohad Ben-Cohen <ohad@wizery.com>
Date: Wed, 30 Nov 2011 15:58:00 +0200
Subject: [PATCH 1/5] rsc_table: publish support for VIRTIO_RING_F_REMOTEPROC

This way virtio will know we're running on a remote processor, which
requires mandatory barriers.

Signed-off-by: Ohad Ben-Cohen <ohad@wizery.com>
---
 src/ti/resources/rsc_table.h |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/ti/resources/rsc_table.h b/src/ti/resources/rsc_table.h
index 7349eb5..a52c45e 100644
--- a/src/ti/resources/rsc_table.h
+++ b/src/ti/resources/rsc_table.h
@@ -120,10 +120,18 @@
 #define VIRTIO_ID_RPMSG		7 /* virtio remote processor messaging */
 
 /* Indices of rpmsg virtio features we support */
-#define VIRTIO_RPMSG_F_NS	0 /* RP supports name service notifications */
+/*
+ * We support name service notifications
+ */
+#define VIRTIO_RPMSG_F_NS               0
+/*
+ * We run on a remote processor, so always use mandatory memory barriers.
+ */
+#define VIRTIO_RING_F_REMOTEPROC	30
 
 /* flip up bits whose indices represent features we support */
-#define IPU_C0_FEATURES         1
+#define IPU_C0_FEATURES         ((1 << VIRTIO_RPMSG_F_NS) | \
+				 (1 << VIRTIO_RING_F_REMOTEPROC))
 
 /* Resource info: Must match include/linux/remoteproc.h: */
 #define TYPE_CARVEOUT    0
-- 
1.6.1

